components/pedigree/ui/src/main/resources/PhenoTips/PedigreeImageService.xml

```diff
@@ -37,13 +37,29 @@
   <minorEdit>false</minorEdit>
   <syntaxId>xwiki/2.1</syntaxId>
   <hidden>true</hidden>
-  <content>{{include reference="PhenoTips.PedigreeMacros" /}}
-
+  <content>
 {{velocity}}
+#if ("$!request.action" == 'export')
+  {{html wiki="false" clean="false"}}
+  $response.setContentType('application/json')##
+  #if ("$!request.image" == '')
+    $response.setStatus(404)##
+    {"error" : "$services.localization.render('phenotips.PedigreeImageService.noPedigreeImage')"}
+    #break
+  #end
+  #set ($pedigreeRasterUrl = $services.svg.rasterizeToTemporaryResource($request.image))
+  #if ("$!request.getParameter('force-download')" == '1')
+    #set ($discard = $pedigreeRasterUrl.getParameters().put('force-download', '1'))
+  #end
+  {"url": "$pedigreeRasterUrl.serialize()"}
+  {{/html}}
+  #break
+#end
 #if ($xcontext.action != 'get')
   **Usage**: //http[s]:~/~/&lt;hostname&gt;[:&lt;port&gt;]/&lt;appname&gt;/bin/get/PhenoTips/PedigreeImageService?[e]id=&lt;identifier&gt;[&amp;format=raster]//
   #break
 #end
+{{include reference="PhenoTips.PedigreeMacros" /}}
```

Comment:
Including `PedigreeMacros` shouldn't produce any extra markup, since it uses `output=false`, so the extra markup comes from the macros.